velodyne_driver/src/lib/input.cc

```diff
@@ -264,24 +265,26 @@ namespace velodyne_driver
   /** @brief constructor
    *
    *  @param private_nh ROS private handle for calling node.
-   *  @param port UDP port number
-   *  @param packet_rate expected device packet frequency (Hz)
-   *  @param filename PCAP dump file name
+   *  @param devip Device IP address.
+   *  @param port UDP port number.
+   *  @param packet_rate expected device packet frequency (Hz).
+   *  @param filename PCAP dump file name.
+   *  @param read_once Read the input file only once.
+   *  @param read_fast Output the data from the input file as fast as possible.
+   *  @param repeat_delay Seconds to wait between repeating input file.
    */
-  InputPCAP::InputPCAP(rclcpp::Node * private_nh, uint16_t port,
-                       double packet_rate, std::string filename):
-    Input(private_nh, port),
+  InputPCAP::InputPCAP(rclcpp::Node * private_nh, const std::string & devip, uint16_t port,
+                       double packet_rate, const std::string & filename, bool read_once, bool read_fast, double repeat_delay):
+    Input(private_nh, devip, port),
     packet_rate_(packet_rate),
-    filename_(filename)
+    filename_(filename),
+    pcap_(nullptr),
+    read_once_(read_once),
+    read_fast_(read_fast),
+    repeat_delay_(repeat_delay)
   {
-    pcap_ = NULL;
     empty_ = true;
 
-    // get parameters using private node handle
-    private_nh->get_parameter("read_once", read_once_);
-    private_nh->get_parameter("read_fast", read_fast_);
-    private_nh->get_parameter("repeat_delay", repeat_delay_);
-
```

Comment:
I'm all for making this class ROS-independent but removing these `get_parameter` calls as well as the ones at lines 82/83 above removes the ability to pass these in as startup parameters. Can you please find a high-level way to gather these and pass them into the constructor? I know they're getting passed as dynamic reconfigure params but they should be taken as launch params as well and the launch params should override the DR params.

Reply:
It is somewhat subtle, but this is still configurable during startup.  In Dashing, the `declare_parameters` call actually declares the parameters *and* gets the current value.  The "current value" is defined as either what is passed in on the command-line during startup, or the default value to the `declare_parameters` call (as seen [here](https://github.com/ros-drivers/velodyne/pull/270/files#diff-5707fd8c8886aa445fb461efc9d17a2fR72)).

You can see this in action by cloning https://github.com/clalancette/declare , then building.  If you run without arguments, it will print 9.0:

```
ros2 run declare declare_node
```

If you run it pointing it to the params yaml file, you'll see 18.0 printed:

```
ros2 run declare declare_node __params:=src/declare/config/declare_params.yaml
```

Let me know if that assuages your concerns here.